File: ios/ScreenShieldRN.m
//
//  ScreenShieldRN.m
//  VeWorld
//
//  Created by Vasileios  Gkreen on 09/05/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import "React/RCTLog.h"

@interface RCT_EXTERN_MODULE(ScreenShieldRN, NSObject)

RCT_EXTERN_METHOD(protectScreenRecording)
RCT_EXTERN_METHOD(protectScreenShot)

@end

File: ios/ScreenShieldRN.swift
//
//  ScreenShieldRN.swift
//  VeWorld
//
//  Created by Vasileios  Gkreen on 09/05/24.
//

import Foundation
import UIKit

@objc(ScreenShieldRN)
class ScreenShieldRN: UIViewController {
  
  
    @objc func protectScreenRecording() {
        ScreenShield.shared.protectFromScreenRecording()
    }
  

    @objc static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
